software/teams/teams.go

```diff
@@ -46,3 +52,75 @@ func Get(teamID string, usersEnabled bool, client houston.ClientInterface, out i
 
 	return nil
 }
+
+// retrieves all teams
+func List(client houston.ClientInterface, out io.Writer) error {
+	var teams []houston.Team
+	var cursor string
+	count := -1
+
+	for len(teams) < count || count == -1 {
+		resp, err := client.ListTeams(cursor, ListTeamLimit)
+		if err != nil {
+			return err
+		}
+		count = resp.Count
+		teams = append(teams, resp.Teams...)
+		cursor = teams[len(teams)-1].ID
+	}
+
+	teamsTable := printutil.Table{
+		Padding:        []int{50, 50},
+		DynamicPadding: true,
+		Header:         []string{"TEAM ID", "TEAM NAME"},
+		ColorRowCode:   [2]string{"\033[1;32m", "\033[0m"},
+	}
+	for i := range teams {
+		teamsTable.AddRow([]string{teams[i].ID, teams[i].Name}, false)
+	}
+	return teamsTable.Print(out)
+}
+
+func Update(teamID, role string, client houston.ClientInterface, out io.Writer) error {
+	if role == "" {
+		fmt.Fprintln(out, noRoleSpecifiedMsg)
+		return nil
+	} else if role != houston.SystemAdminRole && role != houston.SystemEditorRole && role != houston.SystemViewerRole && role != houston.NoneTeamRole {
+		return fmt.Errorf("invalid role: %s, should be one of: %s, %s, %s or %s", role, houston.SystemAdminRole, houston.SystemEditorRole, houston.SystemViewerRole, houston.NoneTeamRole) //nolint:goerr113
+	}
+
+	if role == houston.NoneTeamRole {
+		// Get current role for the team
+		team, err := client.GetTeam(teamID)
+		if err != nil {
+			return err
+		}
+
+		for idx := range team.RoleBindings {
+			if team.RoleBindings[idx].Role == houston.SystemAdminRole || team.RoleBindings[idx].Role == houston.SystemEditorRole || team.RoleBindings[idx].Role == houston.SystemViewerRole {
```

Comment:
It would be nice to have a utility function for that `func IsSystemRole(role string) bool`

Reply:
make sense, have updated the code